client/tests/e2e/specs/file_viewers.spec.ts

```diff
@@ -458,3 +458,121 @@ msTest('Text viewer', async ({ documents }) => {
+  await expect(documents.locator('.sidebar')).toBeVisible();
+
+  // Open file viewer
+  const entries = documents.locator('.folder-container').locator('.file-list-item');
+  await entries.nth(2).dblclick();
+  await expect(documents.locator('.ms-spinner-modal')).toBeHidden();
+  await expect(documents).toBeViewerPage();
+
+  // Sidebar should be auto-hidden in file viewer
+  await expect(documents.locator('.sidebar')).toBeHidden();
+  await expect(documents.locator('#connected-header .topbar')).toBeHidden();
+});
+
+msTest('File viewer sidebar can be toggled in large display', async ({ documents }) => {
+  const entries = documents.locator('.folder-container').locator('.file-list-item');
+  await entries.nth(2).dblclick();
+  await expect(documents).toBeViewerPage();
+
+  // Sidebar initially hidden
+  await expect(documents.locator('.sidebar')).toBeHidden();
+
+  const toggleButton = documents.locator('.file-handler-topbar-buttons__item.toggle-menu');
+  const toggleSidebarButton = documents.locator('#trigger-toggle-menu-button');
+
+  await expect(toggleButton).toBeVisible();
+  await expect(toggleButton).toContainText('Show menu');
+
+  // Show menu and sidebar
+  await toggleButton.click();
+  await expect(documents.locator('#connected-header .topbar')).toBeVisible();
+  await expect(toggleSidebarButton).toBeVisible();
+  await expect(toggleButton).toContainText('Hide menu');
+
+  // Toggle sidebar specifically
+  await toggleSidebarButton.click();
+  await expect(documents.locator('.sidebar')).toBeVisible();
+
+  await toggleSidebarButton.click();
+  await expect(documents.locator('.sidebar')).toBeHidden();
+});
+
+msTest('File viewer sidebar state persists between files', async ({ documents }) => {
+  const entries = documents.locator('.folder-container').locator('.file-list-item');
+
+  // Open first file
+  await entries.nth(2).dblclick();
+  await expect(documents).toBeViewerPage();
+
+  const toggleButton = documents.locator('.file-handler-topbar-buttons__item.toggle-menu');
+  const toggleSidebarButton = documents.locator('#trigger-toggle-menu-button');
+
+  // Show header and sidebar
+  await toggleButton.click();
+  await expect(documents.locator('#connected-header .topbar')).toBeVisible();
+
+  await toggleSidebarButton.click();
+  await expect(documents.locator('.sidebar')).toBeVisible();
+
+  // Navigate to another file via sidebar
+  const sidebar = documents.locator('.sidebar');
+  const recentDocs = sidebar.locator('#sidebar-files').locator('.list-sidebar-content').getByRole('listitem');
+
+  if ((await recentDocs.count()) > 1) {
+    await recentDocs.nth(1).click();
+    await expect(documents.locator('.ms-spinner-modal')).toBeHidden();
+
+    // Sidebar should remain visible
+    await expect(documents.locator('.sidebar')).toBeVisible();
+    await expect(documents.locator('#connected-header .topbar')).toBeVisible();
+  }
+});
+
+msTest('File viewer restores previous sidebar state on exit', async ({ documents }) => {
```

Comment:
Is the other state tested? If the sidebar is visible, is it correctly restored when exiting the viewer?